### ISSUE TITLE
DocTypeGridMapper export fails with NullReferenceException 

### ISSUE THREAD

**Issue by tormnator** [KevinJump/uSync8#416]
**Describe the bug**
Export of content with a DocTypeGridEditor (DTGE) fails if the DTGE contains an image (Umbraco.MediaPicker) and the image has not been set. This bug probably started showing up after a [recent change](https://github.com/KevinJump/uSync/commit/382e96e5dc7fc9094cf8482c3f390e22c934b7e3#diff-2f2632eb9da6f1014826a87be8bbfd2f627a4b2da31a97bba5dcddcda211cb2e) in UdiPickerMapper where the [NullableMapper] attribute was added. This change causes the code in [SyncValueMapperBase.GetExportValue()](https://github.com/KevinJump/uSync/blob/1e7517b46b7d151cf3cafd345ced17011629b0f7/uSync.Core/Mapping/SyncValueMapperBase.cs#L40) to return null in some cases. And that causes the code in [DocTypeGridMapper.GetExportJsonValues()](https://github.com/KevinJump/uSync/blob/v10/dev/uSync.Community.Contrib/Mappers/DocTypeGridMapper.cs#L101) to throw a NullReferenceException on GetExportValue().ToString() since GetExportValue() might return a null value.

**Comment by tormnator** [comment]
I created a fork of uSync and removed the ".ToString() part on line 101 of GetExportJsonValues() since the GetExportValue() method already returned a string. This way a null return value no longer causes a NullReferenceException. I then copied the recompiled uSync.Community.Contrib dll into my Umbraco site's bin folder, ran the site, and successfully exported all the content.

Question, are there any instructions for how to properly work with/contribute to/test a forked/local uSync repository? I assume my approach above isn't the best way. I tried to uninstall the Nuget packages and reference the local uSync dll's, but that didn't work (the Settings section in the back office was messed up). Do I need to create some local/private Nuget package and install it?

**Comment by KevinJump** [comment]
Hi, 

I think these might need a bit of a tidy - as we removed some of the site/build in stuff when it went to v10

For testing i have an Umbraco site in the solution (it's not in source control, maybe it should be) that references the uSync projects: 

```xml
    <ItemGroup>
      <ProjectReference Include="..\uSync.Backoffice.Assets\uSync.Backoffice.Assets.csproj" />
      <ProjectReference Include="..\uSync.BackOffice\uSync.BackOffice.csproj" />
      <ProjectReference Include="..\uSync.Community.Contrib\uSync.Community.Contrib.csproj" />
      <ProjectReference Include="..\uSync.Community.DataTypeSerializers\uSync.Community.DataTypeSerializers.csproj" />
      <ProjectReference Include="..\uSync.Core\uSync.Core.csproj" />
    </ItemGroup>
```

Then when you build this, all the files ect are automatically included in the site so you can run it while you develop

you can also build the nuget packages from the `dist` folder  - there is a build script .build-package takes a major version and build version. 

```
.\build-package 10.2.0 build.20220815.1 
```

if you have it - it should copy to a c:\source\localgit folder, which i then have as a nuget source so i can build new sites with the nuget files.

**Comment by KevinJump** [comment]
Hi for refence this issue is fixed in 10.3 (https://github.com/KevinJump/uSync/releases/tag/v10.3.0)
